clarity_ext/service/file_service.py

```diff
@@ -160,62 +171,61 @@ def upload_files(self, file_handle, files, stdout_max_lines=50):
+    def upload(self, file_handle, instance_name, content):
         """
         :param file_handle: The handle of the file in the Clarity UI
         :param instance_name: The name of this particular file
         :param content: The content of the file. Should be a string.
         """
         artifact = utils.single([shared_file for shared_file in self.artifact_service.shared_files()
                                  if shared_file.name == file_handle])
-        self._upload_single(artifact, file_handle, instance_name, content, stdout_max_lines)
+        self._upload_single(artifact, file_handle, instance_name, content)
 
-    def _upload_single(self, artifact, file_handle, instance_name, content,
-                       stdout_max_lines=50):
+    def _upload_single(self, artifact, file_handle, instance_name, content):
+        """Queues the file for update. Call commit to send to the server."""
         local_path = self.save_locally(content, instance_name)
         self.logger.info("Uploading local file '{}' to the LIMS placeholder at {}".format(
             local_path, file_handle))
 
-        if self.disable_commits:
-            # When not connected to an actual server, we copy the file to
-            # another directory for integration tests
-            upload_path = os.path.join(self.upload_dir, "uploaded")
-            self.logger.info(
-                "disable_commits is on, copying the file to {}".format(upload_path))
-            if not self.os_service.exists(upload_path):
-                self.os_service.mkdir(upload_path)
-            fake_name = "{}_{}".format(
-                artifact.id, os.path.basename(instance_name))
-            new_file_path = os.path.join(upload_path, fake_name)
-            self.os_service.copy_file(local_path, new_file_path)
-        else:
-            self.logger.info("Uploading to the LIMS server")
-            self.os_service.attach_file_for_epp(local_path, artifact)
-
-        if self.uploaded_to_stdout:
-            print("--- {} => {} ({})".format(local_path, artifact.name, artifact.id))
-            extra = 0
-            with self.os_service.open_file(local_path, 'r') as f:
-                for ix, line in enumerate(f):
-                    if ix < stdout_max_lines:
-                        sys.stdout.write(line)
-                    else:
-                        extra += 1
-            if extra > 0:
-                sys.stdout.write("<{} more lines>\n".format(extra))
-            print("---")
+        self.logger.info("Queuing file '{}' for upload to the server".format(local_path))
+        file_name = os.path.basename(local_path)
+        if not file_name.startswith(artifact.id):
+            file_name = "{}_{}".format(artifact.id, file_name)
+        upload_path = os.path.join(self.upload_queue_path, file_name)
+        self.os_service.copy_file(local_path, upload_path)
+
+    def close_local_shared_files(self):
+        for f in self._local_shared_files:
+            f.close()
```

Comment:
Isn't it better to use context managers for this?

Reply:
Yes I think so too. But I believe (I wrote this a long time ago) that the idea here is that the caller should not have to know that things need to be closed, but that it would happen in the destructor of the fileservice object. But it would be cool to merge the ideas.